File: goblinframework-core/src/test/java/org/goblinframework/core/conversion/ConversionServiceTest.java
package org.goblinframework.core.conversion;

import org.goblinframework.core.container.SpringManagedBean;
import org.goblinframework.test.runner.GoblinTestRunner;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.test.context.ContextConfiguration;

import java.time.Instant;
import java.util.Calendar;
import java.util.Date;

import static org.junit.Assert.*;

@RunWith(GoblinTestRunner.class)
@ContextConfiguration("/UT.xml")
public class ConversionServiceTest extends SpringManagedBean {

  @Test
  public void longToCalendar() {
    ConversionService cs = ConversionService.INSTANCE;
    assertTrue(cs.canConvert(Long.class, Calendar.class));
    long millis = System.currentTimeMillis();
    Calendar calendar = cs.convert(millis, Calendar.class);
    assertNotNull(calendar);
    assertEquals(millis, calendar.getTimeInMillis());
  }

  @Test
  public void longToDate() {
    ConversionService cs = ConversionService.INSTANCE;
    assertTrue(cs.canConvert(Long.class, Date.class));
    long millis = System.currentTimeMillis();
    Date date = cs.convert(millis, Date.class);
    assertNotNull(date);
    assertEquals(millis, date.getTime());
  }

  @Test
  public void longToInstant() {
    ConversionService cs = ConversionService.INSTANCE;
    assertTrue(cs.canConvert(Long.class, Instant.class));
    long millis = System.currentTimeMillis();
    Instant instant = cs.convert(millis, Instant.class);
    assertNotNull(instant);
    assertEquals(millis, instant.toEpochMilli());
  }
}

File: goblinframework-core/src/test/java/org/goblinframework/core/conversion/ConversionServiceTest.kt
package org.goblinframework.core.conversion

import org.junit.Assert.*
import org.junit.Test
import java.util.*

class ConversionServiceTest {

  @Test
  fun getNativeConversionService() {
    val cs = ConversionService.INSTANCE.getNativeConversionService()
    assertNotNull(cs)
  }

  @Test
  fun calendarToLong() {
    val cs = ConversionService.INSTANCE
    assertTrue(cs.canConvert(Calendar::class.java, Long::class.java))
    val source = Calendar.getInstance()
    val target = cs.convert(source, Long::class.java)
    assertEquals(source.timeInMillis, target!!.toLong())
  }

  @Test
  fun dateToLong() {
    val cs = ConversionService.INSTANCE
    assertTrue(cs.canConvert(Date::class.java, Long::class.java))
    val source = Date()
    val target = cs.convert(source, Long::class.java)
    assertEquals(source.time, target!!.toLong())
  }
}